subxt/src/rpc/rpc.rs

```diff
@@ -382,6 +366,19 @@ impl<T: Config> Rpc<T> {
         Ok(bytes)
     }
 
+    /// Execute a runtime API call.
+    pub async fn state_call_decoded<Res: Decode>(
+        &self,
+        function: &str,
+        call_parameters: Option<&[u8]>,
+        at: Option<T::Hash>,
+    ) -> Result<Res, Error> {
+        let bytes = self.state_call(function, call_parameters, at).await?;
+        let cursor = &mut &bytes[..];
+        let res: Res = Decode::decode(cursor)?;
+        Ok(res)
+    }
```

Comment:
I'm on the fence about exposing `state_call` and `state_call_decoded` in the RPC interface (just to keep it as slim as possible; it's quite nice imo if each method corresponds to one actual RPC call; in the past this interface has been "muddier" and has had a mix of aggregated calls and "simple" ones, but then it becomes unclear what logic to put in here vs in the higher level interfaces :))

I'd probably just pick one to keep and ditch the other for that reason :)

Reply:
I liked the interface of the `state_call_decoded` more, so I now renamed it to `state_call`. It saves us decoding in multiple places, while there is just one place (`call_raw()`) that requires raw data. I hope the interface change does not break much, I fixed all the tests it affects.